app/main.py

```
@@ -35,10 +38,49 @@ async def refill_tokens():
+        ('end_date', datetime.datetime.fromtimestamp(now - 1800).isoformat(timespec='seconds')),
+        ('transaction_status', 'S'),#Success
+        ('fields', 'all'),
+    ))
+
+@app.on_event("startup")
+@repeat_every(seconds=60)
+async def process_subscriptions():
+    if not ENABLE_PAYPAL:
+        return
+    sql_connection.ping()
+    sql_connection.cursor().execute("UPDATE chat_users.users SET maximum_remaining_messages=(SELECT COUNT(aid)+25 FROM chat_users.subscriptions WHERE subscriptions.user_id=users.user_id AND ? BETWEEN subscription.from_datetime AND subscription.to_datetime AND product IN('RECHARGELIMIT', 'REWARD_RECHARGELIMIT'))", [now])
```

Ruff (0.12.2)

Undefined name `now`

(F821)